src/harness/sas.py

```diff
@@ -304,70 +314,83 @@ def InjectCpiUser(self, request):
     _RequestPost('https://%s/admin/injectdata/cpi_user' % self._base_url,
                  request, self._tls_config)
 
-  def TriggerCreateActivityDump(self, request):
-    _RequestPost('https://%s/admin/trigger/create_full_activity_dump' % self._base_url,
-                 request, self._tls_config)
-
-  def InjectPeerSas(self,request):
-    _RequestPost('https://%s/admin/injectdata/peer_sas' % self._base_url,
-                 request, self._tls_config)
-
+  def TriggerLoadDpas(self):  
+    _RequestPost('https://%s/admin/trigger/load_dpas' %
+                 self._base_url, None, self._tls_config)
+    
+  def TriggerBulkDpaActivation(self, request):
+    _RequestPost('https://%s/admin/trigger/bulk_dpa_activation' %
+                 self._base_url, request,
+                 self._GetDefaultAdminSSLCertPath(),
+                 self._GetDefaultAdminSSLKeyPath())
+
+  def TriggerDpaActivation(self, request):
+    _RequestPost('https://%s/admin/trigger/dpa_activation' %
+                 self._base_url, request,
+                 self._GetDefaultAdminSSLCertPath(),
+                 self._GetDefaultAdminSSLKeyPath()) 
+
+  def TriggerDpaDeactivation(self, request):
+    _RequestPost('https://%s/admin/trigger/dpa_deactivation' %
+                 self._base_url, request,
+                 self._GetDefaultAdminSSLCertPath(),
+                 self._GetDefaultAdminSSLKeyPath()) 
+    
   def _GetDefaultAdminSSLCertPath(self):
     return os.path.join('certs', 'admin_client.cert')
 
   def _GetDefaultAdminSSLKeyPath(self):
     return os.path.join('certs', 'admin_client.key')
 
-def GetServer():
-  config_parser = ConfigParser.RawConfigParser()
-  config_parser.read(['sas.cfg'])
-  server_base_url = config_parser.get('HttpServer', 'BaseUrl')
-  server_port = int(config_parser.get('HttpServer', 'Port'))
-  return HttpServer({'baseUrl': server_base_url, 'port': server_port})
 
-class HttpServer(sas_interface.HttpServerInterface):
-  """Test Harness acting as a Http Server to receive Pull/GET and Push/POST
+class HttpServer(httpserver_interface.HttpServerInterface):
+  """Test Harness acting as a Http Server to receive Pull/GET
    Requests from SAS Under Test
+   parameters: is the dict variable for different types of dump objects.
+   schema of parameters: { "full_activity" : dict to store full activity dump objects,
```

Comment:
I know we only have one full activity dump object. I think we also need to packaging this full activity dump object into parameter (because this full activity dump object needed to be fetched by SAS UUT). So for the parameters I give a key called "full_activity", value will be the full activity dump object.

Reply:
Also, do you mean it is better to write a helper function to generate FAD object automatically? Should we put the helper function into util.py?

Reply:
I was suggesting that this function can create the FAD object based on the other keys+values in this dictionary. Since the helper function is specific to this class/function, it's probably best to keep it in this class.